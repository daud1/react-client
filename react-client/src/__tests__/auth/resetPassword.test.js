import React from 'react';
import { shallow, render, mount } from 'enzyme';
import ResetPassword from '../../components/auth/resetPassword.component';
import shallowToJson from 'enzyme-to-json';
import moxios from 'moxios';


describe ('ResetPassword Component', () => {
    it('should render without crashing', () => {
        const wrapper = shallow(<ResetPassword />);
        expect(wrapper.exists(<div></div>)).toBe(true);
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<ResetPassword />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('handles successful axios calls properly', (done) => {
        const wrapper = shallow(<ResetPassword />);

        wrapper.find('input[name="new_password"]').simulate('change', { target: { value: 'pswd' } });
        wrapper.find('input[name="confirm"]').simulate('change', { target: { value: 'pswd' } });
        wrapper.find('form').simulate('submit', {preventDefault() {} });

        moxios.stubRequest('http://localhost:5000/auth/reset-password', {
            status: 200,
            'MSG': 'Password successfully reset!'
        });
        moxios.wait(() => {
            expect(wrapper.html()).toContain('Password successfully reset!');
            done();
        });
    });
})