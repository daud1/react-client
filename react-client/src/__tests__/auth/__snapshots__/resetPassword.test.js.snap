// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword Component matches snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      placeholder="Enter your new password here."
      type="password"
    />
    <input
      placeholder="Type password again to confirm."
      type="password"
    />
    <button
      type="submit"
    >
      Reset Password
    </button>
  </form>
</div>
`;
